### ISSUE TITLE
collectionView dynamic collection assignment

### ISSUE THREAD

**Issue by adam-26** [lazojs/lazo#271]
Hi,

The lib/public/views/collection.js (collectionView) requires a collection be assigned before constructor is called, this makes it impossible to dynamically assign a collection to the view on creation. 

Is it possible to modify the order of operations during view construction so that a collection can be assigned dynamically using the initialize function (or similar)? Looking at the code, I think this will require changes in flexo.

Thanks,

Adam.

**Comment by jstrimpel** [comment]
Would changing the property resolution in the constructor to use `_.result` meet your use case needs?

**Comment by adam-26** [comment]
I don't think so. My use case requires access to the views controller to determine the collection to use, I don't believe this is accessible until after the constructor has completed.

**Comment by jstrimpel** [comment]
I see. Would the following suffice as a permanent fix or at the minimum a workaround?

```javascript
define(['lazoView'], function (LazoView) {
  'use strict';
  
  return LazoView.extend({
    constructor: function (options) {
      this.collection = options.ctl.getCollection();
      LazoView.call(this, options);
    }
  });
});
```

**Comment by adam-26** [comment]
Yes, that works. Thanks!
